code/ATMOSPHERICS/components/components_base.dm

```diff
@@ -199,7 +195,6 @@ Helpers
 */
 
 /obj/machinery/atmospherics/components/proc/update_parents()
-	for(var/P in parents)
-		var/datum/pipeline/parent = parents["[P]"]
-		if (parent && istype(parent))
-			parent.update = 1
\ No newline at end of file
+	for(var/I = 1; I <= device_type; I++)
+		var/datum/pipeline/parent = parents["p[I]"]
+		parent.update = 1
```

Comment:
that was just him putting it back to how I'd done it
the reason I have all the loops like that is basically just for consistency
for a lot of the loops it is necessary (the ones before the lists are initialized, the ones where there's multiple lists being edited) and for the ones for which it isn't necessary, I thought it'd be nice to just let it all be consistent